src/main/native/SymmetricCipher.c

```diff
@@ -449,21 +449,32 @@ JNIEXPORT int CIPHER_zKMC_internal(unsigned char* input, unsigned char* output,
 
 JNIEXPORT jint JNICALL Java_com_ibm_crypto_plus_provider_ock_NativeInterface_z_1kmc_1native(JNIEnv * env, jclass clazz, jbyteArray input, jint inputOffset, jbyteArray output, jint outputOffset, jlong paramPointer, jint inputLength, jint mode)
 {
-  // Get input and output buffer
   jboolean isCopy = 0;
   jint len = 0;
-  unsigned char* inputPointer  = (unsigned char*)((*env)->GetPrimitiveArrayCritical(env, input,  &isCopy)) + inputOffset;
-  unsigned char* outputPointer = (unsigned char*)((*env)->GetPrimitiveArrayCritical(env, output, &isCopy)) + outputOffset;
 
-  if( NULL == outputPointer || NULL == inputPointer ) {
+  // Get input buffer.
+  unsigned char* inputPointer  = (unsigned char*)((*env)->GetPrimitiveArrayCritical(env, input,  &isCopy));
```

Comment:
Do we need the `isCopy` as a parameter, since it's just hardcoded a few lines above?

Reply:
I dont mind having `isCopy` as a variable here given that if that were to change this code ( unlikely ) we would change it in one spot for both the input and output buffers. It also makes it a bit more readable knowing that the value represents if a copy should be made or not in the JNI `GetPrimitiveArrayCritical` calls. Do you see any advantage to hardcoding `&0` here?

Reply:
I don't think we've ever specified this as true in any of our code given that it's easier to pass a copy through Java if needed, rather than having JNI make a copy. Also, the VM might have to make a copy regardless of what you specify if arrays are in different format.

But, in any case, I'm OK with leaving the variable there. But could we at least declared it as `jboolean isCopy = JNI_FALSE` to make it more clear?

Reply:
Yes agreed it is more clear to declare `isCopy` as `JNI_FALSE`. It makes this more readable. Update this using https://github.com/IBM/OpenJCEPlus/pull/294/commits/401a22e22da7b18d16537b840fb63bd16469c5de

Reply:
Also agree that typically we create the copy in the Java code not in JNI and that we dont really know for sure if the VM made a copy or not. We are just requesting that no copy is made here.